File: 0240 - Search a 2D Matrix II/solution.cpp
class Solution {
public:
    bool searchMatrix(vector<vector<int>>& A, int target) {
        int m = A.size(), n = A[0].size(), i = 0, j = n-1;
        while(i<m && j>=0){
            if(A[i][j] == target) return true;
            A[i][j]<target?i++:j--;
        }
        return false;
    }
};

File: 0240 - Search a 2D Matrix II/solution.py
class Solution:
    def searchMatrix(self, A: List[List[int]], target: int) -> bool:
        m, n, i, j = len(A), len(A[0]), 0, len(A[0])-1
        while i < m and j >= 0:
            if A[i][j] == target:
                return True
            elif A[i][j] < target:
                i += 1
            elif A[i][j] > target:
                j -= 1
        return False